bot/exts/help_channels/_cog.py

```diff
@@ -125,14 +125,19 @@ async def claim_channel(self, message: discord.Message) -> None:
         """
         log.info(f"Channel #{message.channel} was claimed by `{message.author.id}`.")
         await self.move_to_in_use(message.channel)
-        await self._handle_role_change(message.author, message.author.add_roles)
 
-        await _message.pin(message)
+        # Handle odd edge case of `message.author` being a `discord.User` (see bot#1839)
+        if isinstance(message.author, discord.User):
```

Comment:
Do you think it would make more sense to invert this i.e. check for the type it supports rather than checking for the one it can't support? Technically the author could be _any_ `abc.User`.

Reply:
I did the `User` check first since then the comment directly correlates to the first statement. It doesn't really matter which way round the statement is, and afaik the only possible subclassess of `abc.User` is `User` and `Member` (can't be `ClientUser`, even when message is sent by the bot).